providers/keap/write.go

```diff
@@ -0,0 +1,83 @@
+package keap
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/spyzhov/ajson"
+)
+
+func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*common.WriteResult, error) {
+	if err := config.ValidateParams(); err != nil {
+		return nil, err
+	}
+
+	url, err := c.getWriteURL(config.ObjectName)
+	if err != nil {
+		return nil, err
+	}
+
+	var write common.WriteMethod
+
+	if len(config.RecordId) == 0 {
+		if supportedObjectsByCreate[c.Module.ID].Has(config.ObjectName) {
+			write = c.Client.Post
+		}
+	} else {
+		if supportedObjectsByUpdatePUT[c.Module.ID].Has(config.ObjectName) {
+			write = c.Client.Put
+
+			url.AddPath(config.RecordId)
+		}
+
+		if supportedObjectsByUpdatePATCH[c.Module.ID].Has(config.ObjectName) {
+			write = c.Client.Patch
+
+			url.AddPath(config.RecordId)
+		}
+	}
+
+	if write == nil {
+		// No supported REST operation was found for current object.
+		return nil, common.ErrOperationNotSupportedForObject
+	}
+
+	res, err := write(ctx, url.String(), config.RecordData)
+	if err != nil {
+		return nil, err
+	}
+
+	body, ok := res.Body()
+	if !ok {
+		return &common.WriteResult{
+			Success: true,
+		}, nil
+	}
+
+	// write response was with payload
+	return constructWriteResult(config, c.Module.ID, body)
+}
+
+func constructWriteResult(
+	config common.WriteParams, moduleID common.ModuleID, body *ajson.Node,
+) (*common.WriteResult, error) {
+	identifier := objectNameToWriteResponseIdentifier[moduleID].Get(config.ObjectName)
+
+	recordID, err := identifier.Find(body)
```

Comment:
This is the main code that locates identifier field in the response body.
Some objects have exceptions and for one case id is under the nested object.